Fix goroutine stall on write to broken socket

funny-falcon-at-joomcode opened the pull request:
imho, combination of
```go
	_, w.err = w.buffers.WriteTo(w.c)
	if w.err != nil {
		w.buffers = nil
	}
	w.cond.Broadcast()
```
and
```go
	for len(w.buffers) != 0 {
		w.cond.Wait()
	}
```
certainly have to lead to goroutine stall-ness on any write error.

Attempt to fix it by changing condition on "flush count".

If I'm mistaken, then excuse me for disturbing.

funny-falcon-at-joomcode commented:
I forgot a bit about flow of my thoughts, but looks like you re-found it correctly. Excuse me I didn't describe it thoroughly on PR creation.

And I were care in first place of latency of writer: when other writers win race for lock, one unfortunate writer could loop in this condition for a long time even in case of no errors. Then I found it could stuck forever in certain cases.

Probably "certainly have to lead" is an hyperbole. I was tired at that moment, and probably started to see `!=` as `==`.

martin-sucha commented:
@funny-falcon-at-joomcode could you please `git commit --amend` the commit message and update the pull request description to include the above steps describing the race condition?

Also could you please add a commit adding yourself to the `AUTHORS` file as described in [CONTRIBUTING.md](https://github.com/gocql/gocql/blob/master/CONTRIBUTING.md#minimum-requirement-checklist)?

I think once that is done @Zariel or @alourie will merge this pull request.

funny-falcon-at-joomcode commented:
Well, looking more closely I think I were mistaken: there is no way for goroutine to stall.
But still current wait loop condition could lead to larger resources utilization then it should be.
I've changed commit message, and added myself to AUTHORS
